types/auth.go

```diff
@@ -7,5 +7,11 @@ type AuthConfig struct {
 	Auth          string `json:"auth,omitempty"`
 	Email         string `json:"email"`
 	ServerAddress string `json:"serveraddress,omitempty"`
+
+	// IdentityToken is used to authenticate the user and get
+	// an access token for the registry.
+	IdentityToken string `json:"identitytoken,omitempty"`
```

Comment:
Sorry I'm late to this party, but I do feel that @calavera comment in https://github.com/docker/docker/pull/20970#issuecomment-192535355 does seem to make sense. Do we need a new field if `IdentityToken` and `Password` are mutually exclusive? I wouldn't fight for it much if there's a nice benefit to do it this way. Thanks!